functions/src/application/portal.ts

```diff
@@ -302,3 +303,31 @@ export const get_developer_profile = async (request: Request, response: Response
     });
   }
 };
+
+export const create_vanity_link = async (req: Request, res: Response) => {
+  const { first_name, last_name, email, destination, primary_domain, subdomain, slashtag } = req.body;
+  let error = null;
+
+  await build_vanity_link_v2({
+    first_name,
+    last_name,
+    email,
+    destination,
+    primary_domain,
+    subdomain,
+    slashtag,
+  }).catch((err) => {
+    error = [err.response.data];
+  });
+
+  if (!error)
+    return res.status(201).send({
+      message: "Sucessfully created Vanity link",
+      url: `https://${subdomain}.${primary_domain}/${slashtag}`,
+    });
+
+  return res.status(400).send({
+    message: "Failed to create Vanity link",
+    error,
+  });
```

Comment:
Actually, I take back what I said about lines 323-332. This is a perfect use case for try-catch with async/await. See this [guide](https://javascript.info/async-await#error-handling), and restructure the code so you're not creating a dummy `error` variable.

Reply:
+1 ^^, use try/catch... there's plenty of examples scattered throughout the codebase. Also log the error with sentry in the catch block.

 - [Example 1](https://github.com/acmutd/portal-backend/blob/0f3652bec44e52bb48e7d9aa7c96b90a2b7cb0a6/functions/src/custom/event.ts#L80)
 - [Example 2](https://github.com/acmutd/portal-backend/blob/0f3652bec44e52bb48e7d9aa7c96b90a2b7cb0a6/functions/src/mail/sendgrid.ts#L50)
 - [Example 3](https://github.com/acmutd/portal-backend/blob/0f3652bec44e52bb48e7d9aa7c96b90a2b7cb0a6/functions/src/application/typeform.ts#L207)